pkg/generate/app/cmd/template.go

```diff
@@ -38,9 +38,11 @@ func TransformTemplate(tpl *templateapi.Template, client client.TemplateConfigsN
 
 	// ensure the template objects are decoded
 	// TODO: in the future, this should be more automatic
-	if errs := runtime.DecodeList(result.Objects, kapi.Codecs.UniversalDecoder()); len(errs) > 0 {
-		err = errors.NewAggregate(errs)
-		return nil, fmt.Errorf("error processing template %s: %v", name, err)
+	if decode {
+		if errs := runtime.DecodeList(result.Objects, kapi.Codecs.UniversalDecoder()); len(errs) > 0 {
```

Comment:
I think the invalid output is caused by the k8s printing machinery (`kubectl.GetPrinter`/`kubectl.NewVersionedPrinter`) which doesn't know about `User` objects as it is currently initialized.

It's probably better not to use `DecodeList` in `oc process` to keep the output format. For example it currently outputs
```yaml
apiVersion: v1
items:
- apiVersion: v1
  kind: User
  metadata:
    name: somename
kind: List
metadata: {}
```

However if it used `DecodeList` (and changes were made to print `User` objects correctly) the output would look like this, with extra entries that weren't in the original template:
```yaml
apiVersion: v1
items:
- apiVersion: v1
  groups: null
  identities: null
  kind: User
  metadata:
    creationTimestamp: null
    name: somename
kind: List
metadata: {}
```